Router not support multipart put document with attachment

Step to reproduce:
1. Clone https://github.com/couchbaselabs/photo-drop
2. Run the app on an android device and an iOS device
3. Send a photo from iOS to android.

Exception:
```
05-07 19:25:25.845    8997-9090/android.jamiltz.com.photodrop W/Router﹕ WARNING: Exception parsing body into dictionary
    com.fasterxml.jackson.core.JsonParseException: Unexpected character ('-' (code 45)) in numeric value: expected digit (0-9) to follow minus sign, for valid numeric value
            at [Source: Acme.Serve.Serve$ServeInputStream@11341df6; line: 2, column: 3]
            at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1486)
            at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:518)
            at com.fasterxml.jackson.core.base.ParserBase.reportUnexpectedNumberChar(ParserBase.java:991)
            at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._handleInvalidNumberStart(UTF8StreamJsonParser.java:2610)
            at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._parseNegNumber(UTF8StreamJsonParser.java:1366)
            at com.fasterxml.jackson.core.json.UTF8StreamJsonParser._nextTokenNotInObject(UTF8StreamJsonParser.java:784)
            at com.fasterxml.jackson.core.json.UTF8StreamJsonParser.nextToken(UTF8StreamJsonParser.java:697)
            at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:3600)
            at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3545)
            at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2646)
            at com.couchbase.lite.router.Router.getBodyAsDictionary(Router.java:172)
            at com.couchbase.lite.router.Router.do_PUT_Document(Router.java:1795)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.couchbase.lite.router.Router.start(Router.java:471)
            at com.couchbase.lite.listener.LiteServlet.service(LiteServlet.java:128)
            at javax.servlet.http.HttpServlet.service(HttpServlet.java:802)
            at Acme.Serve.Serve$ServeConnection.runServlet(Serve.java:2251)
            at Acme.Serve.Serve$ServeConnection.parseRequest(Serve.java:2176)
            at Acme.Serve.Serve$ServeConnection.run(Serve.java:1988)
            at Acme.Utils$ThreadPool$PooledThread.run(Utils.java:1223)
            at java.lang.Thread.run(Thread.java:818)
05-07 19:25:25.848    8997-9090/android.jamiltz.com.photodrop E/Router﹕ Router unable to route request to do_PUT_Document
    java.lang.reflect.InvocationTargetException
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.couchbase.lite.router.Router.start(Router.java:471)
            at com.couchbase.lite.listener.LiteServlet.service(LiteServlet.java:128)
            at javax.servlet.http.HttpServlet.service(HttpServlet.java:802)
            at Acme.Serve.Serve$ServeConnection.runServlet(Serve.java:2251)
            at Acme.Serve.Serve$ServeConnection.parseRequest(Serve.java:2176)
            at Acme.Serve.Serve$ServeConnection.run(Serve.java:1988)
            at Acme.Utils$ThreadPool$PooledThread.run(Utils.java:1223)
            at java.lang.Thread.run(Thread.java:818)
     Caused by: com.couchbase.lite.CouchbaseLiteException
            at com.couchbase.lite.router.Router.do_PUT_Document(Router.java:1797)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.couchbase.lite.router.Router.start(Router.java:471)
            at com.couchbase.lite.listener.LiteServlet.service(LiteServlet.java:128)
            at javax.servlet.http.HttpServlet.service(HttpServlet.java:802)
            at Acme.Serve.Serve$ServeConnection.runServlet(Serve.java:2251)
            at Acme.Serve.Serve$ServeConnection.parseRequest(Serve.java:2176)
            at Acme.Serve.Serve$ServeConnection.run(Serve.java:1988)
            at Acme.Utils$ThreadPool$PooledThread.run(Utils.java:1223)
            at java.lang.Thread.run(Thread.java:818)
```

--- a/src/main/java/com/couchbase/lite/Database.java
+++ b/src/main/java/com/couchbase/lite/Database.java
@@ -405,14 +405,6 @@ public final class Database {
         //recursively delete attachments path
         boolean deleteAttachmentStatus = FileDirUtils.deleteRecursive(attachmentsFile);
 
-        //recursively delete path where attachments stored( see getAttachmentStorePath())
-        int lastDotPosition = path.lastIndexOf('.');
-        if( lastDotPosition > 0 ) {
-            File attachmentsFileUpFolder = new File(path.substring(0, lastDotPosition));
-            FileDirUtils.deleteRecursive(attachmentsFileUpFolder);
-        }
-
-
         if (!deleteStatus) {
             throw new CouchbaseLiteException("Was not able to delete the database file", Status.INTERNAL_SERVER_ERROR);
         }

